crates/flox-rust-sdk/src/models/publish.rs

```diff
@@ -290,6 +311,27 @@ impl<'flox> Publish<'flox, NixAnalysis> {
         Ok(())
     }
 
+    pub async fn check_substituter(
+        &mut self,
+        substituter: SubstituterUrl,
+    ) -> Result<(), PublishError> {
+        let cache_meta = self.get_binary_cache_metadata(Some(substituter)).await?;
+        let cache_meta_json =
+            serde_json::to_value(cache_meta).map_err(PublishError::SerializeCacheMeta)?;
+        let mut caches = self
+            .analysis
+            .0
+            .get("cache")
+            .map(|v| v.as_array().expect("invalid metadata").to_owned())
+            .unwrap_or_default();
+
+        caches.push(cache_meta_json);
```

Comment:
What was blocking us on using catalog types? This whole method could essentially be dropped

Reply:
I think mainly the custom flakeref representation we have there that is kind of everywhere in the tests and such.
I'd prefer to use the catalog entry type otherwise yes

Reply:
https://github.com/flox/flox/issues/203